src/viam/module/module.py

```diff
@@ -79,6 +80,12 @@ async def start(self):
         """Start the module service and gRPC server"""
         try:
             await self.server.serve(log_level=self._log_level, path=self._address)
+        except SystemExit as e:
```

Comment:
what does this do? maybe leave a comment around this bit

Reply:
comments added

Reply:
It does not resolve RSDK-2551, and I'm actually not sure we should add this @njooma (totally open to other opinions). The except here doesn't actually catch anything from grpclib. It's confusing, but the only error you can catch from grpclib's second stage of graceful exit is `asyncio.CancelledError`. From what I can tell, there's no way to catch the raised `SystemExit(143)`.

Reply:
Removed